File: Code-Chef/Distinct-Digit-Number/distinct_digit_number.py
'''
Problem Statement: https://www.codechef.com/HCS12019/problems/SETH0006
Author: striker
'''

def is_distinct_digits(number):
    freq_digits = [False] * 10
    while number:
        if freq_digits[number % 10]:
            return False
        freq_digits[number % 10] = True
        number = int(number / 10)
    return True

def main():
    start,end = tuple(map(int,input().split()))
    assert(start <= end)
    while start <= end:
        if is_distinct_digits(start):
            print(start)
            break
        start += 1
    if start > end:
        print("-1")

if __name__ == "__main__":
    main()

File: Code-Chef/Distinct-Digit-Number/distinct_digit_number.c
/*  Problem Statement: https://www.codechef.com/HCS12019/problems/SETH0006
    Author: striker
*/

#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include<string.h>
#include<assert.h>

#define NOT_FOUND -1

const bool is_distinct_digits(size_t);

int main(void) {
    size_t start,end;
    scanf("%lu%lu",&start,&end);
    assert((start > 0 && start < 100001) && (end > 0 && end < 100001));
    while(start <= end) {
        if(is_distinct_digits(start)) {
            printf("%lu\n", start);
            break;
        }
        ++start;
    }
    if(start > end) {
        printf("%d\n",NOT_FOUND);
    }
    return 0;
}

const bool is_distinct_digits(size_t n) {
    bool is_all_digit_distinct = true, freq_digits[10];;
    size_t temp = n;
    memset(freq_digits,false,(10 * sizeof(bool)));
    while(temp) {
        if(freq_digits[temp % 10]) {
            is_all_digit_distinct = false;
            break;
        }
        freq_digits[temp % 10] = true;
        temp /= 10;
    }
    return is_all_digit_distinct;
}